pyiceberg/io/pyarrow.py

```diff
@@ -1407,7 +1408,7 @@ def list(self, list_type: ListType, list_array: Optional[pa.Array], value_array:
                 # This can be removed once this has been fixed:
                 # https://github.com/apache/arrow/issues/38809
                 list_array = pa.LargeListArray.from_arrays(list_array.offsets, value_array)
-
+            value_array = self._cast_if_needed(list_type.element_field, value_array)
```

Comment:
These fixes are necessary to ensure that the types within ListTypes and MapTypes are also promoted on write

Reply:
Ah good catch, we have those on the map, but the list is missing indeed 👍